sample-apps/discount-functions-sample-app/web/index.js

```diff
@@ -0,0 +1,406 @@
+import { join } from "path";
+import { readFileSync } from "fs";
+import express from "express";
+import cookieParser from "cookie-parser";
+import { Shopify, LATEST_API_VERSION } from "@shopify/shopify-api";
+import dotenv from "dotenv";
+
+import applyAuthMiddleware from "./middleware/auth.js";
+import verifyRequest from "./middleware/verify-request.js";
+import { setupGDPRWebHooks } from "./gdpr.js";
+import redirectToAuth from "./helpers/redirect-to-auth.js";
+import { BillingInterval } from "./helpers/ensure-billing.js";
+import { AppInstallations } from "./app_installations.js";
+import { idToGid } from "./helpers/gid.js";
+import { METAFIELD } from "./helpers/metafield.js";
+
+const USE_ONLINE_TOKENS = false;
+
+const PORT = parseInt(process.env.BACKEND_PORT || process.env.PORT, 10);
+
+// TODO: There should be provided by env vars
+const DEV_INDEX_PATH = `${process.cwd()}/frontend/`;
+const PROD_INDEX_PATH = `${process.cwd()}/frontend/dist/`;
+
+const DB_PATH = `${process.cwd()}/database.sqlite`;
+
+const {
+  SHOPIFY_ORDER_DISCOUNT_ID,
+  SHOPIFY_PRODUCT_DISCOUNT_ID,
+  SHOPIFY_SHIPPING_DISCOUNT_ID,
+} = dotenv.parse(readFileSync(join(process.cwd(), "../", ".env"), "utf8"));
+
+Shopify.Context.initialize({
+  API_KEY: process.env.SHOPIFY_API_KEY,
+  API_SECRET_KEY: process.env.SHOPIFY_API_SECRET,
+  SCOPES: process.env.SCOPES.split(","),
+  HOST_NAME: process.env.HOST.replace(/https?:\/\//, ""),
+  HOST_SCHEME: process.env.HOST.split("://")[0],
+  API_VERSION: LATEST_API_VERSION,
+  IS_EMBEDDED_APP: true,
+  // This should be replaced with your preferred storage strategy
+  // See note below regarding using CustomSessionStorage with this template.
+  SESSION_STORAGE: new Shopify.Session.SQLiteSessionStorage(DB_PATH),
+  ...(process.env.SHOP_CUSTOM_DOMAIN && {
+    CUSTOM_SHOP_DOMAINS: [process.env.SHOP_CUSTOM_DOMAIN],
```

Comment:
Just a question, but does SPIN update this env here? Or do we need to set that?

Reply:
I wouldn't expect this to be populated by Spin, but we can confirm by printing out the value in your instance: `echo $SHOP_CUSTOM_DOMAIN`

Reply:
Forgot to mention that you need to run that inside the `cli` repo, like you normally do when setting up functions
https://vault.shopify.io/page/Using-Functions-in-Spin~s7J5.md